app/builders/account_builder.rb

```diff
@@ -2,11 +2,13 @@
 
 class AccountBuilder
   include CustomExceptions::Account
-  pattr_initialize [:account_name!, :email!, :confirmed!]
+  pattr_initialize [:account_name!, :email!, :confirmed!, :user]
 
   def perform
-    validate_email
-    validate_user
+    if @user.nil?
+      validate_email
+      validate_user
+    end
     ActiveRecord::Base.transaction do
       @account = create_account
       @user = create_and_link_user
```

Comment:
let's return both the `account` and `user` from this builder. So we can render it both from the builder for the account creation method. Hence you will be able to access the newly created account along with the user in request response